tests/unittest_scoped_nodes.py

```diff
@@ -1499,13 +1507,10 @@ class A(object): pass
     def test_metaclass_lookup_inference_errors(self):
         module = builder.parse(
             """
-        import six
-
         class Metaclass(type):
             foo = lala
 
-        @six.add_metaclass(Metaclass)
-        class B(object): pass
+        class B(object, metaclass=Metaclass): pass
```

Comment:
I do not understand why in some tests `six` is removed whereas in other it is kept.

Reply:
`SixBrainTest` keeps six because it is meant to work with six.  But other than that one-off exception, my understanding is that the other tests used six to implement 2/3 compatibility within the unit test itself. Python 2 and six support was dropped from the codebase in an earlier commit, but wasn't fully purged from the unit tests, likely because six was still getting installed by tox. The purpose of this PR is to purge all these forgotten uses of six which were likely just oversights from the earlier PR that dropped Python 2.

Reply:
@dgilman i understand and agree with you. However what about the `test_metaclass_generator_hack` which is not part of `SixBrainTest` but keep using `six`? Two other tests (`test_using_invalid_six_add_metaclass_call`, `test_with_metaclass_mro`) are in the same case but given the name or commit comment i'm ok to keep `six` in them.

Reply:
See the comment in scoped_nodes.py:1684 . This is the "metaclass hack" that this is all referring to. In order to create a metaclass that exercises these code paths you have to jump through a bunch of hoops, which is what [six does in its with_metaclass()](https://github.com/benjaminp/six/blob/master/six.py#L840) . Going back to the scoped_nodes.py function, you see it tests for `self.name == "with_metaclass"` before applying the hack, which is six's name for the function that is doing all of this. So the unit test and code conditionals were written with six in mind, so it makes sense to keep the unit test relying on six.

Reply:
@dgilman thanks for this clear explanation and sorry for all those questions.